### PR TITLE
fix: Change localhost to 127.0.0.1. Fixes #3564 (#3564)

### LINKED ISSUE #3564: [bug] localhost not point to 127.0.0.1
https://github.com/argoproj/argo/blob/13b1d3c19e94047ae97a071e4468b1050b8e292b/server/apiserver/argoserver.go#L232

`endpoint` should be explicit set as `127.0.0.1` but `localhost`. There are cases `localhost` is not resolved to 127.0.0.1

### PR DESCRIPTION
**Pull request by lonsdale8734**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Comment by sarabala1979** [comment]
Can you make sure code lint correctly by running `make lint`

**Comment by lonsdale8734** [comment]
> Can you fix code lint by running `make lint` or `make pre-commit`?

I am new here for the Argo Community. Is there any guides to build local development environment with docker?

**Comment by sarabala1979** [comment]
https://github.com/argoproj/argo/blob/master/docs/CONTRIBUTING.md
https://github.com/argoproj/argo/blob/master/docs/running-locally.md

File: server/apiserver/argoserver.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ import (
 const (
 	// MaxGRPCMessageSize contains max grpc message size
 	MaxGRPCMessageSize = 100 * 1024 * 1024
+	Localhost = "127.0.0.1"
```
Comment (sarabala1979) [inline server/apiserver/argoserver.go:52]: Do you really need a constant which is used only on place?  you can do it inline.

**Comment by sarabala1979** [comment]
@lonsdale8734 will you able to set up local env and do make lint?

**Comment by alexec** [comment]
No activity in 4 weeks. Marking as `wontfix`. Please dismiss reviews when you want this to be reviewed again.

**Pull request closed without merging**